templates/modelExpt/run_modelComp_expt.m

```diff
@@ -0,0 +1,170 @@
+function expt = run_modelComp_expt(expt,bTestMode)
+% RUN_MODELCOMP_EXPT - A simple AAF compensation experiment where F1 is
+% shifted up on 20% of trials, F1 shifted down on 20%, and no shift 60%.
+%
+% Set bTestMode input argument to 1 to run a shortened version.
+%
+% Based around run_cerebAAF_expt and run_modelExpt_expt
+
+% v1 2023-10 - Chris Naber
+
+%% Default arguments
+
+if nargin < 1, expt = []; end
+if nargin < 2 || isempty(bTestMode), bTestMode = 0; end
+
+%% Set up general experiment parameters
+expt.name = 'modelComp';
+expt.trackingFileLoc = 'experiment_helpers'; % Where the OST/PCF files are kept (for audapter_viewer)
+expt.trackingFileName = 'measureFormants'; % What the files are called (does not include Working/Master)
+refreshWorkingCopy(expt.trackingFileLoc,expt.trackingFileName);
+if ~isfield(expt,'snum'), expt.snum = get_snum; end
+expt.dataPath = get_acoustSavePath(expt.name,expt.snum);
+if ~exist(expt.dataPath,'dir')
+    mkdir(expt.dataPath)
+end
+
+if ~isfield(expt,'gender'), expt.gender = get_height; end       % vocal tract length or gender correlate for LPC Order
+
+% In other experiments, instead of assuming LPC, we collect some
+%   speech samples and allow the experimenter to evaluate different LPCs on
+%   that dataset. For simplicity here, LPC is just set arbitrarily.
+if strcmp(expt.gender, 'male')
+    expt.audapterParams.nLPC = 17;
+else
+    expt.audapterParams.nLPC = 15;
+end
+
+%perturbation amount, in mels
+shiftMag = 125;
+
+% timing
+expt.timing.stimdur = 1.9;         % time stim is on screen, in seconds
+expt.timing.interstimdur = .75;    % minimum time between stims, in seconds
+expt.timing.interstimjitter = .75; % maximum extra time between stims (jitter)
+expt.timing.visualfbdur = 0.75; 
+
+%set up duration feedback parameters
+expt.durcalc.min_dur = .4; % set_exptDefaults normally sets to .25
+expt.durcalc.max_dur = .65;
+expt.durcalc.ons_thresh = 0.15;
+expt.durcalc.offs_thresh = 0.4;
+
+%% Set up duration practice
+exptDur = expt;
+exptDur.session = 'dur'; 
+exptDur.dataPath = fullfile(expt.dataPath,exptDur.session);
+if ~exist(exptDur.dataPath, 'dir')
+    mkdir(exptDur.dataPath)
+end
+exptDur.ntrials = 6;
+
+exptDur.shiftMags   = zeros(1,exptDur.ntrials);
+exptDur.shiftAngles = zeros(1,exptDur.ntrials);
+
+exptDur.words = {'head' 'dead' 'Ted'};
+exptDur.allWords = mod(0:exptDur.ntrials-1, numel(exptDur.words)) + 1;
```

Comment:
Do we want these to be in cyclic order like this?

Reply:
TODO it doesn't need to be that complicated -- I'll simplify that line and make it random order